src/message_broadcaster/mod.rs

```diff
@@ -1,16 +1,45 @@
 use eyre::Result;
+use serde::{Deserialize, Serialize};
+use ssz_derive::{Decode, Encode};
 
 use crate::price_provider::Price;
-use bls_signatures::Signature;
+use bls::{PublicKey, Signature};
 
+pub mod json;
 pub mod log;
 
-#[derive(Debug)]
-pub struct PriceMessage {
+#[derive(Debug, Serialize, Deserialize)]
+pub struct OracleMessage {
+    pub value_message: SignedPriceValueMessage,
+    pub interval_inclusion_messages: Vec<SignedIntervalInclusionMessage>,
+    pub validator_public_key: PublicKey,
+}
+
+#[derive(Debug, Decode, Encode, Serialize, Deserialize)]
+pub struct PriceValueMessage {
     pub price: Price,
+    pub slot_number: u64,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct SignedPriceValueMessage {
+    pub message: PriceValueMessage,
+    pub signature: Signature,
+}
+
+#[derive(Debug, Decode, Encode, Serialize, Deserialize)]
+pub struct IntervalInclusionMessage {
+    pub value: u64,
+    pub interval_size: u64,
```

Comment:
Since `interval_size` and `slot_number` is constant for each round I guess we could factor them out of here and add them once to the top level of the message to reduce the amount of data we will be sending. (Although we would want to keep them in the individual messages when signing them).

Reply:
On the other hand it will make verifying signatures slightly more complex on the server side, so we could also just leave it like this and change it if we encounter issues with the messages being too large.

Reply:
Yes. Simple first please 🙏. Optimize only as needed or if the code would be equally complex.